packages/jest-resolve-dependencies/src/index.js

```diff
@@ -50,24 +39,29 @@ class DependencyResolver {
     if (!dependencies) {
       return [];
     }
-    return compact(
-      dependencies.map(dependency => {
+
+    return dependencies
+      .map(dependency => {
         if (this._resolver.isCoreModule(dependency)) {
           return null;
         }
         try {
           return this._resolver.resolveModule(file, dependency, options);
         } catch (e) {}
-        return this._resolver.getMockModule(file, dependency) || null;
-      }),
-    );
+        return this._resolver.getMockModule(file, dependency);
+      })
+      .filter(Boolean);
```

Comment:
How big `dependencies` array may get? Did you measure `filter` vs `for-loop` perf here?

Reply:
`filter` is 1.5-2x faster than `for-loop` in node.js 9.11.1 on my macbook.
https://gist.github.com/Connormiha/638a5f4ba7a711573ee98800615487cb

Reply:
Thanks!